components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/sync/WorkManagerSyncManagerTest.kt

```diff
@@ -34,4 +56,42 @@ class WorkManagerSyncManagerTest {
         assertNull(getSyncState(testContext))
         assertEquals(0L, getLastSynced(testContext))
     }
+
+    @Test
+    fun `Given debounce is set to false THEN sync time always expires`() {
+        `when`(mockTags.contains(SyncWorkerTag.Debounce.name)).thenReturn(false)
+
+        engineSyncTimestamp["test"] = System.currentTimeMillis() - SYNC_STAGGER_BUFFER_MS - 100L
+        engineSyncTimestamp["test2"] = System.currentTimeMillis()
+
+        val workerManagerSyncWorker = WorkManagerSyncWorker(testContext, mockParam)
+
+        assertFalse(workerManagerSyncWorker.isDebounced())
+        assertFalse(workerManagerSyncWorker.lastSyncedWithinStaggerBuffer("test"))
```

Comment:
Little bit of a rant incoming. No need for changes, just some thoughts that might be worth sharing.

Generally speaking, I prefer to focus on tests that cover the smallest possible public API. This creates expectations about the behavior of the class as it would actually be used. Marking functions as internal in order to test them makes a little more sense in a modular architecture like we have in A-C, as the `internal` keyword is still respected across module boundaries. I find it to be a bit of an anti-pattern in Fenix, though, where the entire project is still a single module and all those `internal` functions become part of the global namespace. 

That said, the existing structure of this code is really hard to test well. That's always a bit of a problem when using Android platform types like `WorkManager`, but I think we could make pretty huge improvements in that regard if we get the chance to refactor more of this in the future. So any increase in test coverage here is a good thing in my books 🙂

Reply:
Agreed.  The first time I did similar thing for testing I was uncomfortable as well.  I wish tests by default can call into `private` functions.